include/clap/version.h

```diff
@@ -20,11 +20,18 @@ typedef struct clap_version {
 }
 #endif
 
-#define CLAP_VERSION_MAJOR ((uint32_t)1)
-#define CLAP_VERSION_MINOR ((uint32_t)1)
-#define CLAP_VERSION_REVISION ((uint32_t)5)
+#define CLAP_VERSION_MAJOR 1
+#define CLAP_VERSION_MINOR 1
+#define CLAP_VERSION_REVISION 5
+
 #define CLAP_VERSION_INIT                                                                          \
-   { CLAP_VERSION_MAJOR, CLAP_VERSION_MINOR, CLAP_VERSION_REVISION }
+   { (uint32_t)CLAP_VERSION_MAJOR, (uint32_t)CLAP_VERSION_MINOR, (uint32_t)CLAP_VERSION_REVISION }
+
+#define CLAP_VERSION_LT(maj,min,rev) ((maj < CLAP_VERSION_MAJOR) || \
+                    (maj == CLAP_VERSION_MAJOR && min < CLAP_VERSION_MINOR ) || \
+                    (maj == CLAP_VERSION_MAJOR && min == CLAP_VERSION_MINOR && rev < CLAP_VERSION_REVISION))
+#define CLAP_VERSION_EQ(maj,min,rev) ((maj == CLAP_VERSION_MAJOR) && (min == CLAP_VERSION_MINOR) && (rev == CLAP_VERSION_REVISION))
+#define CLAP_VERSION_GE(maj,min,rev) (!CLAP_VERSION_LT(maj,min,rev))
```

Comment:
Do we really need LT and EQ?

Reply:
`LT=!GE` right?

Reply:
I think they're useful to have, and having them around doesn't really cost you anything.  `CLAP_VERSION_LT(x, y z)` has mcuh clearer semantics than writing out `!CLAP_VERSION_GE(x, y, z)` yourself. `EQ` is really specific of course but I can see it being useful if we ever ship a single borked CLAP revision.

Reply:
With JUCE I've used all the operators including range comparisons. Which is why a number is nice but if we are going with macros, yeah, with < and == you can get everything; and GE is there just since it is a common case.

but we can do whatever you want. We can also provide no macros if you prefer and users can just write them themselves if they need, once the uint is stripped from the version. Up to you.

Reply:
In your macro, you need brackets around `(maj) < CLAP_VERSION_MAJOR`

Reply:
> `LT=!GE` right?

I mean, the code 4 lines above your comemnt shows indeed, that GE == !LT, since that's how I coded it :)

Reply:
> I mean, the code 4 lines above your comemnt shows indeed, that GE == !LT, since that's how I coded it :)

Yeah sorry, my comment was poorly formulated, I wanted to say that if we have GE we don't need anything else I suppose, but I've removed that comment, unfortunately you were too fast ;-)

Reply:
Pushed the bracket change.